### ISSUE TITLE
Use black `--preview` for formatting.

### ISSUE THREAD

**Issue by mcrumiller** [pola-rs/polars#12849]
### Description

Edit: looks like we don't use black anyway, but ruff's formatter which is a close equivalent, but doesn't yet support black's style. Not sure if this is feasible at the moment.

---

The [black preview](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html) style can be enabled with the `preview = true` flag in our `pyproject.toml` file. Enabling this would (in my opinion) make a lot of the code more readable. Primarily, the [improved multiline dictionary and list indentation](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-multiline-dictionary-and-list-indentation-for-sole-function-parameter) makes this:

```python
pl.DataFrame(
    {
        "a": [1, 2, 3],
        "b": [4, 5, 6],
    }
)
```
into this:
```python
pl.DataFrame({
    "a": [1, 2, 3],
    "b": [4, 5, 6],
})
```
Which is *much* more concise and removes an entire level of indentation.

The drawback is of course that the preview style is subject to change, but I think the pros outweigh the cons.

**Comment by cmdlineluser** [comment]
I think ruff just added this: https://github.com/astral-sh/ruff/pull/8293

**Comment by stinodego** [comment]
That would match what `rustfmt` does so I would be in favor. When `ruff` releases this along with their docstring formatter, we can have a nice big diff :)

**Comment by stinodego** [comment]
We cannot use the docstring formatter yet because there is a bug in it. It was fixed and I hope to be able to use it when `0.1.9` comes out.

Regarding the preview style: we will just patiently wait for it to become stable, which should happen in ruff `0.2.0`. There is no rush with this and we don't want to mess with the formatting more than needed. So I'll close this issue, but we will pick this up automatically when upgrading ruff.
